repository/src/test/java/org/alfresco/repo/content/directurl/ContentStoreDirectUrlConfigUnitTest.java

```diff
@@ -0,0 +1,234 @@
+
+        assertTrue("Expected content store direct URLs to be enabled", contentStoreDirectUrlConfig.isEnabled());
+        contentStoreDirectUrlConfig.validate();
+        assertFalse("Expected content store direct URLs to be disabled", contentStoreDirectUrlConfig.isEnabled());
+    }
+
+    @Test
+    public void testInvalidConfig_MaxExpiryTimeNegative()
+    {
+        setupDirectAccessConfig(ENABLED, DEFAULT_EXPIRY_TIME_IN_SECS, -1L);
+
+        assertTrue("Expected content store direct URLs to be enabled", contentStoreDirectUrlConfig.isEnabled());
+        contentStoreDirectUrlConfig.validate();
+        assertFalse("Expected content store direct URLs to be disabled", contentStoreDirectUrlConfig.isEnabled());
+    }
+
+    @Test
+    public void testInvalidConfig_MaxExpiryTimeExceedsSystemMax()
+    {
+        Long maxExpiryTimeInSec = contentStoreDirectUrlConfig.getSysWideMaxExpiryTimeInSec() + 1;
+        setupDirectAccessConfig(ENABLED, DEFAULT_EXPIRY_TIME_IN_SECS, maxExpiryTimeInSec);
+
+        assertTrue("Expected content store direct URLs to be enabled", contentStoreDirectUrlConfig.isEnabled());
+        contentStoreDirectUrlConfig.validate();
+        assertFalse("Expected content store direct URLs to be disabled", contentStoreDirectUrlConfig.isEnabled());
+    }
+
+    /* Helper method to set content store direct access url configuration settings */
+    private void setupDirectAccessConfig(Boolean isEnabled, Long defaultExpiryTime, Long maxExpiryTime)
+    {
+        contentStoreDirectUrlConfig.setIsEnabled(isEnabled);
+        contentStoreDirectUrlConfig.setDefaultExpiryTimeInSec(defaultExpiryTime);
+        contentStoreDirectUrlConfig.setMaxExpiryTimeInSec(maxExpiryTime);
+    }
+
+    /* Helper method to verify content store direct access url configuration settings */
+    private void verifyDirectAccessConfig(Boolean isEnabled, Long defaultExpiryTime, Long maxExpiryTime)
+    {
+        assertEquals("Expected content store direct URLs to be enabled = " + isEnabled, isEnabled, contentStoreDirectUrlConfig.isEnabled());
+        assertEquals("Expected default expiry time to match " + defaultExpiryTime, defaultExpiryTime, contentStoreDirectUrlConfig.getDefaultExpiryTimeInSec());
+        assertEquals("Expected maximum expiry time to match " + maxExpiryTime, maxExpiryTime, contentStoreDirectUrlConfig.getMaxExpiryTimeInSec());
+    }
+
+    /* Helper method to set system-wide direct access url configuration settings */
+    private void setupSystemWideDirectAccessConfig()
+    {
+        SystemWideDirectUrlConfig sysConfig = new SystemWideDirectUrlConfig();
+        sysConfig.setIsEnabled(ENABLED);
+        sysConfig.setDefaultExpiryTimeInSec(SYS_DEFAULT_EXPIRY_TIME_IN_SECS);
+        sysConfig.setMaxExpiryTimeInSec(SYS_MAX_EXPIRY_TIME_IN_SECS);
+        sysConfig.validate();
+        contentStoreDirectUrlConfig.setSystemWideDirectUrlConfig(sysConfig);
+    }
+}
```

Comment:
Tests for multiple content stores?

Reply:
Might it be better to test multiple content stores in the integration tests so that we can have multiple implementations and/or mutiple stores with the same config?